repo/content/content_upgrade_lock.go

```diff
@@ -122,11 +127,42 @@ func (l *UpgradeLock) totalDrainInterval() time.Duration {
 	return l.MaxPermittedClockDrift + 2*l.IODrainTimeout
 }
 
+func (l *UpgradeLock) checkCoordinatorLock(ctx context.Context) (bool, error) {
+	req, err := http.NewRequestWithContext(ctx, "GET", l.CoordinatorURL, http.NoBody)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to prepare coordinator lock request with %q", l.CoordinatorURL)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to check for coordinator lock with %q", l.CoordinatorURL)
+	}
+	defer resp.Body.Close() //nolint:errcheck
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		return true, nil
+	case http.StatusLocked:
```

Comment:
not sure here, 409 feels more appropriate, as 423 belongs to WebDAV.

Reply:
will change to 409